### PR TITLE
bump version of oauth2-proxy, add flag to add custom prover CA

### PR DESCRIPTION
**Pull request by stroebitzer**
**What this PR does / why we need it**:

Customers should be able to add their custom CA.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

oauth2-proxy needs a version push to support the flag `--provider-ca-file`

**Documentation**:
https://oauth2-proxy.github.io/oauth2-proxy/configuration

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

**Comment by xrstf** [comment]
@stroebitzer please bump the IAP chart version number.

File: charts/iap/values.yaml [hunk 1]
```diff
@@ -15,9 +15,11 @@
 iap:
   image:
     repository: quay.io/oauth2-proxy/oauth2-proxy
-    tag: v6.0.0
+    tag: v6.1.1
     pullPolicy: IfNotPresent
 
+  # custom_provider_ca_secret: provider-ca-secret
```
Comment (xrstf) [inline charts/iap/values.yaml:22]: Please add a comment line to explain what this does.

File: charts/iap/values.yaml [hunk 2]
```diff
@@ -15,9 +15,11 @@
 iap:
   image:
     repository: quay.io/oauth2-proxy/oauth2-proxy
-    tag: v6.0.0
+    tag: v6.1.1
```
Comment (xrstf) [inline charts/iap/values.yaml:18]: Please also adjust the appVersion in the Chart.yaml.

**Review by xrstf — CHANGES_REQUESTED** [review]
Good idea with a few minor leftovers. :)

**Comment by xrstf** [comment]
@stroebitzer Ping :)

File: charts/iap/values.yaml (hunk 1 above)
Comment (stroebitzer) [reply charts/iap/values.yaml:22]: added comment, thx for hint

File: charts/iap/values.yaml (hunk 2 above)
Comment (stroebitzer) [reply charts/iap/values.yaml:18]: updated version, thx for hint

**Review by xrstf — APPROVED** [review]
/approve

**Comment by toschneck** [comment]
@xrstf can we please backport this PR and #6080 to the `2.15` release branch, somehow needed more often :-)

**Pull request merged**